plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1918,18 +1931,41 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
 
         try {
             TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriter(session, handle.getLocation());
-
-            long createdTime = Instant.now().toEpochMilli();
             transactionLogWriter.appendCommitInfoEntry(getCommitInfoEntry(session, commitVersion, createdTime, SET_TBLPROPERTIES_OPERATION, readVersion));
             transactionLogWriter.appendProtocolEntry(protocolEntry);
 
+            metadataEntry.ifPresent(transactionLogWriter::appendMetadataEntry);
+
             transactionLogWriter.flush();
         }
         catch (IOException e) {
             throw new TrinoException(DELTA_LAKE_BAD_WRITE, "Failed to write Delta Lake transaction log entry", e);
         }
     }
 
+    public MetadataEntry buildMetadataEntry(ConnectorSession session, DeltaLakeTableHandle handle, Map<String, String> configuration, long createdTime)
+    {
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, handle);
+        List<String> partitionColumns = getPartitionedBy(tableMetadata.getProperties());
+        List<DeltaLakeColumnHandle> columns = tableMetadata.getColumns().stream()
+                .filter(columnMetadata -> !columnMetadata.isHidden())
+                .map(columnMetadata -> toColumnHandle(columnMetadata, columnMetadata.getName(), columnMetadata.getType(), partitionColumns))
+                .collect(toImmutableList());
+        return new MetadataEntry(
```

Comment:
Can we create a helper method to get the `MetadataEntry`? and we can use that helper method in `appendTableEntries` as well?

Reply:
I tried but it became messy, I will do it as another PR, ok ?